### PR TITLE
[Coverity] GPU Plugin Medium issues

### PR DESCRIPTION
**Pull request by mhpanah**
### Details:
 - This PR fixes: 1562334, 1559879, 1559863, 1559854, 1559677, 1559879, 1559863

- 1559461 and 1559460 are related to ov::pass::pattern::matcher in `decompose_reduce_scalar_output.cpp`
- 1559886 and 1559881 are related to ov::pass::pattern::matcher in  `transpose_fusion.cpp`
- 1559862  is related to ov::pass::pattern::matcher in `transpose_fusion.cpp`
- 1559698 and 1559691 are related to ov::pass::pattern::matcher in `bcast_and_pad_zp_buffers.cpp`

- all member variables are initialized in the constructors' initializer lists of  `ov::pass::pattern::matcher` to address 1559461, 1559460, 1559886, 1559881, 1559862, 1559698, 1559691.

### Tickets:
 - [*CVS-153064*](https://jira.devtools.intel.com/browse/CVS-153064)
 - CVS-145082

### CODE REVIEW

**Comment by p-durandin** [comment]
buid_jenkins

File: src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp [hunk 1]
```diff
@@ -757,6 +757,11 @@ void FullyConnected_bf_tiled::GetUpdateDispatchDataFunc(KernelData& kd) const {
             const auto& prim_params = static_cast<const fully_connected_params&>(params);
 
             size_t quantize_grp_size = get_dynamic_quantize_group_size(prim_params);
+            if (quantize_grp_size == 0) {
+                std::cerr << "Error: quantize_grp_size is zero." << std::endl;
+                return;
+            }
```
Comment (p-durandin) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp:763]: Please use exception or assert here

File: src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp [hunk 2]
```diff
@@ -941,6 +946,11 @@ KernelsData FullyConnected_bf_tiled::GetMultiKernelsData(const Params &params,
 
     size_t quantize_grp_size = get_dynamic_quantize_group_size(fc_params);
 
+    if (quantize_grp_size == 0) {
+        std::cerr << "Error: quantize_grp_size is zero." << std::endl;
+        return KernelsData();
```
Comment (p-durandin) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp:951]: Please use exception or assert here

**Review by p-durandin — APPROVED** [review]

**Comment by p-durandin** [comment]
build_jenkins

**Comment by mhpanah** [comment]
build_jenkins

**Comment by p-durandin** [comment]
build_jenkins

**Comment by p-durandin** [comment]
build_jenkins

**Comment by p-durandin** [comment]
build_jenkins

**Pull request merged**